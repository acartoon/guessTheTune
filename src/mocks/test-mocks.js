export const testMocks = [
  {
    answers: [
      {id: `9mT`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `fRl`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `LIz`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `rock`},
      {id: `qJ3`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
    ],
    genre: `jazz`,
    id: `7wO`,
    type: `genre`,
  },
  {
    answers: [
      {id: `Sub`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `aWl`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `BmX`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `ZGz`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `rock`},
    ],
    genre: `pop`,
    id: `30K`,
    type: `genre`,
  },
  {
    answers: [
      {id: `4G1`, picture: `https://api.adorable.io/avatars/134`, artist: `John Snow`},
      {id: `eVm`, picture: `https://api.adorable.io/avatars/102`, artist: `John Snow`},
      {id: `Gey`, picture: `https://api.adorable.io/avatars/120`, artist: `Jack Daniels`},
    ],
    id: `C9G`,
    song: {artist: `John Snow`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`},
    type: `artist`
  },
  {
    answers: [
      {id: `4G1`, picture: `https://api.adorable.io/avatars/134`, artist: `John Snow`},
      {id: `eVm`, picture: `https://api.adorable.io/avatars/102`, artist: `John Snow`},
      {id: `Gey`, picture: `https://api.adorable.io/avatars/120`, artist: `Jack Daniels`},
    ],
    id: `C9G`,
    song: {artist: `John Lenon`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`},
    type: `artist`
  },
];
