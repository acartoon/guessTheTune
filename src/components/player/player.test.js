import React from 'react';
import renderer from 'react-test-renderer';
import {testMocks} from '../../mocks/test-mocks.js';
import Player from './player.jsx';

it(`Player correctly renders after relaunch`, () => {
  const tree = renderer
  .create(<Player
    src={testMocks[2].song.src}
    isPlaying={false}
    onPlayButtonClick={() => null}/>, {
    createNodeMock: () => {
      return {};
    }
  })
  .toJSON();
  expect(tree).toMatchSnapshot();
});
