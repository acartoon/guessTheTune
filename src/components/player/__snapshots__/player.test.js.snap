// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player correctly renders after relaunch 1`] = `
<div
  className="track"
>
  <button
    className="track__button track__button--play"
    disabled={true}
    onClick={[Function]}
    type="button"
  />
  <div
    className="track__status"
  >
    <audio />
  </div>
</div>
`;
